### ISSUE TITLE
Add mechanism in RenderSliver to check pointer status.

### ISSUE THREAD

**Issue by dcov** [flutter/flutter#27516]
I'm trying to create a custom refresh indicator similar to `CupertinoRefreshControl` but that doesn't trigger the refresh action until the user has released the drag. To do this I need to know when the user has released the drag and then determine if they dragged enough to trigger it.

It'd be a lot nicer to be able to implement this with a custom `RenderSliver` rather than have to make a custom `CustomScrollView`.

**Comment by Kartik1607** [comment]
Flutter does that. Check out [Notification class](https://docs.flutter.io/flutter/widgets/Notification-class.html). To get *Scroll* notification you would need to use [Scroll Notification](https://docs.flutter.io/flutter/widgets/ScrollNotification-class.html)

**Comment by dcov** [comment]
Yeah I'm aware of the Scroll-Notification's, but this is a different issue than getting scroll updates at the `Widget` layer. I'm asking for a way to check whether the user pointer is still active i.e. a pointer up event hasn't happened, at the `RenderSliver`/`RenderObject` layer. 

For example it can be a constraint in `SliverConstraints`, similar to `userScrollDirection`, named `userScrollStatus` or something.
